### PR TITLE
[Serializer] Unset object_to_populate after using it

### PR DESCRIPTION
**Pull request by dunglas**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

The `object_to_populate` key must be unset after using it to avoid problems when normalizing sub objects. Needed for #17193.

### CODE REVIEW

File: src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php [hunk 1]
```diff
@@ -279,7 +279,7 @@ protected function prepareForDenormalization($data)
      * Instantiates an object using constructor parameters when needed.
      *
      * This method also allows to denormalize data into an existing object if
-     * it is present in the context with the object_to_populate key.
+     * it is present in the context with the object_to_populate key then remove this key.
```
Comment (jakzal) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: This sentence doesn't sound right anymore.

Comment (dunglas) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: PHPDoc updated.

Comment (jakzal) [inline src/Symfony/Component/Serializer/Normalizer/AbstractNormalizer.php]: Better :+1:

**Comment by fabpot** [comment]
should be merged in 2.7, right?

**Comment by dunglas** [comment]
Yes

**Comment by fabpot** [comment]
Thank you @dunglas.

**Pull request closed without merging**